modules/contact/src/actions/ContactAction.C

```diff
@@ -249,6 +249,45 @@ ContactAction::ContactAction(const InputParameters & params)
   else if (getParam<bool>("ping_pong_protection"))
     paramError("ping_pong_protection",
                "The 'ping_pong_protection' option can only be used with the 'ranfs' formulation");
+
+  // Remove repeated interactions
+  std::vector<std::pair<BoundaryName, BoundaryName>> lean_boundary_pairs;
+
+  for (const auto & [primary, secondary] : _boundary_pairs)
+  {
+    const auto & primary_lambda = primary;
+    const auto & secondary_lambda = secondary;
+
+    auto it = std::find_if(lean_boundary_pairs.begin(),
+                           lean_boundary_pairs.end(),
+                           [&](const std::pair<BoundaryName, BoundaryName> & boundary_pairs)
+                           {
+                             bool exist = false;
+                             const bool match_one = boundary_pairs.second == secondary_lambda &&
+                                                    boundary_pairs.first == primary_lambda;
+                             const bool match_two = boundary_pairs.second == primary_lambda &&
+                                                    boundary_pairs.first == secondary_lambda;
+                             exist = match_one || match_two;
+                             return exist;
+                           });
+
+    if (it == lean_boundary_pairs.end())
+      lean_boundary_pairs.push_back({primary, secondary});
+    else
+      mooseInfo("Contact pair ",
+                primary,
+                "--",
+                secondary,
+                " has been removed from the contact interaction list due to "
+                "duplicates in the input file.");
+  }
+
+  _boundary_pairs = lean_boundary_pairs;
+
+  if (_boundary_pairs.size() == 0)
+    paramError("primary",
+               "Number of contact pairs after removing repeated ones is zero. Please revise your "
```

Comment:
This can only happen if the parameter was empty from the beginning, so I'd put that check upfront and not even mention the removal of repeated pairs.